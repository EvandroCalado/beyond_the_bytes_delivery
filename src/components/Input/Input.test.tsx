import { fireEvent, render, screen } from 'utils';

import { Input } from './Input';

describe('Input', () => {
  it('should render default', () => {
    render(<Input label="test" name="name" type="text" />);

    const label = screen.getByText('test');
    const input = screen.getByRole('textbox');

    expect(label).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'text');
    expect(input).toHaveAttribute('name', 'name');
    expect(input).toHaveValue('');
  });

  it('should render with type email', () => {
    render(<Input label="test" name="email" type="email" />);

    const input = screen.getByRole('textbox');

    expect(input).toHaveAttribute('type', 'email');
  });

  it('should render on change', () => {
    render(<Input label="test" name="name" type="text" />);

    const input = screen.getByRole('textbox');

    expect(input).toHaveValue('');

    fireEvent.change(input, { target: { value: 'test' } });

    expect(input).toHaveValue('test');
  });

  it('should render a snapshot', () => {
    const { container } = render(
      <Input label="test" name="email" type="email" />,
    );

    expect(container).toMatchSnapshot();
  });
});
