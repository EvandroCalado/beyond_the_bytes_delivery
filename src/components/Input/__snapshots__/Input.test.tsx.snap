// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render a snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: .8rem;
}

.c1 {
  font-size: 1.2rem;
  text-transform: capitalize;
  font-weight: 600;
}

.c2 {
  outline: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  border-radius: .8rem;
  border: 1px solid #131118;
}

.c2:focus {
  outline: 3px solid #131118;
  outline-offset: 1px;
}

.c2:disabled {
  opacity: 0.3;
  pointer-events: none;
}

<div>
  <div
    class="c0 input"
  >
    <label
      class="c1"
      for="email"
    >
      test
    </label>
    <input
      class="c2"
      id="email"
      name="email"
      type="email"
    />
  </div>
</div>
`;
